test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_trcomp.py

```diff
@@ -208,5 +208,212 @@ def test_trcomp_debug(self, patched, ecr_image, sagemaker_session, tmpdir, py_ve
+
+        instance_count = 1
+        instance_type = "ml.p3.16xlarge"
+        
+        source_dir = (
+            "./examples/question-answering"
+            if Version(transformers_version) < Version("4.6")
+            else "./examples/pytorch/question-answering"
+        )
+
+        num_gpus_per_instance = 8
+        hyperparameters["max_steps"] = (3 * num_gpus_per_instance * instance_count)
+        
+        with timeout(minutes=DEFAULT_TIMEOUT):
+            estimator = HuggingFace(
+                compiler_config=TrainingCompilerConfig(debug=True),
+                entry_point='run_qa.py',
+                source_dir=source_dir,
+                git_config=git_config,
+                metric_definitions=metric_definitions,
+                role='SageMakerRole',
+                image_uri=ecr_image,
+                instance_count=instance_count,
+                instance_type=instance_type,
+                sagemaker_session=sagemaker_session,
+                hyperparameters=hyperparameters,
+                environment={'GPU_NUM_DEVICES':'8'},
+                py_version=py_version,
+                max_retry_attempts=15,
+            )
+            estimator.fit(job_name=sagemaker.utils.unique_name_from_base('hf-pt-trcomp-SNMG-debug'), logs=True)
+
+        captured = capsys.readouterr()
+        logs = captured.out+captured.err
+        assert "Found configuration for Training Compiler" in logs
+        assert "Training Compiler set to debug mode" in logs
+        assert "Configuring SM Training Compiler" in logs
+        assert "device: xla" in logs
+
+        debug_artifact_path=estimator.model_data.replace('model.tar.gz','output.tar.gz')
+        debug_artifact=os.path.join(tmpdir, 'output.tar.gz')
+        subprocess.check_output(['aws', 's3', 'cp', debug_artifact_path, debug_artifact])
+        with tarfile.open(debug_artifact, 'r:gz') as tarball:
+            tarball.extractall(path=tmpdir)
+        xla_metrics_file = os.path.join(tmpdir, 'compiler', 'XLA_METRICS_FILE.txt')
+        assert os.path.exists(xla_metrics_file)
+
+@pytest.mark.integration("sagmaker-training-compiler")
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_py2_containers
+@pytest.mark.skip_huggingface_containers
+@pytest.mark.skip_cpu
+@mock.patch('sagemaker.huggingface.TrainingCompilerConfig.validate', return_value=None)
+class TestMultiNodeMultiGPU:
+    '''
+    All Multi Node Multi GPU tests go here.
+    '''
+    @pytest.mark.model("bert-large")
+    def test_trcomp_default(self, patched, ecr_image, sagemaker_session, tmpdir, py_version, capsys):
+        '''
+        Tests the default configuration of SM trcomp
+        '''
+        transformers_version = get_transformers_version(ecr_image)
+        git_config = {'repo': 'https://github.com/huggingface/transformers.git', 'branch': 'v'+transformers_version}
+
+        instance_count = 2
+        instance_type = "ml.p4d.24xlarge"
```

Comment:
convert instance count and instance type into fixtures and parametrize the instance type for p3.16xlarge and p4d.24xlarge (without and with EFA)